server/types/date.go

```diff
@@ -246,7 +243,11 @@ func (b DateType) SerializeValue(val any) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return converted.(time.Time).MarshalBinary()
+	t, err := converted.(pgdate.Date).ToTime()
```

Comment:
If we can serialize `pgdate.Date` to/from `time.Time`, then that must mean there's a 1-to-1 correlation between the two types right? In that case, why do we need to use `pgdate.Date` at all?

If that's not true and we should actually be using a bespoke serialization format, then this also affects things such as `SerializedCompare` that assume `time.Time` serialization.